src/main/webapp/availability.js

```diff
@@ -38,3 +38,17 @@ function browserTimezoneOffset() {
 function availabilityTableDiv() {
   return document.getElementById('table-container');
 }
+
+function updateAvailability() {
+  let selectedSlots = document.getElementsByClassName('table-success');
+  let firstSlot = document.getElementsByTagName('tbody').item(0)
+    .firstElementChild.firstElementChild.getAttribute('data-utc');
+  let lastSlot = document.getElementsByTagName('tbody').item(0)
+    .lastElementChild.lastElementChild.getAttribute('data-utc');
+  let utcTimestamps = [
```

Comment:
This system of having the first two elements of the array be the period boundaries is a bit strange and fragile. It will be even more surprising to people reading the Java code. It would probably be better to have an object (which will be mirrored in Java for use with Gson) that has the fields `firstSlot`, `lastSlot`, and `selectedSlots`. You would construct it as:

```
let requestObj = {
  firstSlot: firstSlot,
  lastSlot: lastSlot,
  selectedSlots: selectedSlots,
}
```

Reply:
Done